index/index.go

```diff
@@ -193,59 +121,98 @@ func (index *Index) addFile(subPath string) error {
 	return nil
 }
 
-func newIndexer(path string, fileIndex int, index *Index) indexer {
-	indexer := indexer{
-		index:     index,
-		fileIndex: fileIndex,
-		path:      path,
+func (index *Index) buildIndexOverFile(fileId int, f indexFile) error {
+	entry := make([]byte, 12)
+
+	for {
+		offset, err := f.file.Seek(0, 1)
```

Comment:
One of those args is the argument and the other is a constant saying where to seek from, right? Can you use that constant?